server/service/integration_enterprise_test.go

```diff
@@ -4464,6 +4464,23 @@ func (s *integrationEnterpriseTestSuite) TestRunHostScript() {
 	require.False(t, scriptResultResp.HostTimeout)
 	require.Contains(t, scriptResultResp.Message, fleet.RunScriptAlreadyRunningErrMsg)
 
+	// an async script doesn't care about timeouts
+	now := time.Now()
+	mysql.ExecAdhocSQL(t, s.ds, func(tx sqlx.ExtContext) error {
+		_, err := tx.ExecContext(ctx, `UPDATE host_script_results SET created_at = ? WHERE execution_id = ?`,
+			now.Add(-1*time.Hour),
+			runResp.ExecutionID,
+		)
+		return err
+	})
+	scriptResultResp = getScriptResultResponse{}
+	s.DoJSON("GET", "/api/latest/fleet/scripts/results/"+runResp.ExecutionID, nil, http.StatusOK, &scriptResultResp)
+	require.Equal(t, host.ID, scriptResultResp.HostID)
+	require.Equal(t, "echo", scriptResultResp.ScriptContents)
+	require.Nil(t, scriptResultResp.ExitCode)
+	require.False(t, scriptResultResp.HostTimeout)
+	require.Contains(t, scriptResultResp.Message, fleet.RunScriptAlreadyRunningErrMsg)
```

Comment:
seems like we're returning this message in many cases for async scripts (eg: see the test above, it's a freshly enqueued async script)

the message is later ignored in the UI, but I believe we should adjust the API to return the message only when relevant.

I created https://github.com/fleetdm/fleet/issues/16431